src/background/http-instrument.ts

```diff
@@ -1,687 +1,670 @@
+    // headers, visit_id [auto-filled], time_stamp
+    const update = {} as HttpRequest;
 
-  update.crawl_id = crawlID;
+    update.crawl_id = crawlID;
 
-  // ChannelId is a unique identifier that can be used to link requests and
-  // responses. FF 49+
-  update.channel_id = httpChannel.channelId;
+    // requestId is a unique identifier that can be used to link requests and responses
+    update.channel_id = details.requestId;
 
-  const stacktrace_str = get_stack_trace_str();
-  update.req_call_stack = loggingDB.escapeString(stacktrace_str);
+    // const stacktrace_str = get_stack_trace_str();
+    // update.req_call_stack = escapeString(stacktrace_str);
 
-  const url = httpChannel.URI.spec;
-  update.url = loggingDB.escapeString(url);
+    const url = details.url;
+    update.url = escapeString(url);
 
-  const requestMethod = httpChannel.requestMethod;
-  update.method = loggingDB.escapeString(requestMethod);
+    const requestMethod = details.method;
+    update.method = escapeString(requestMethod);
 
-  let referrer = "";
-  if (httpChannel.referrer) {
-    referrer = httpChannel.referrer.spec;
-  }
-  update.referrer = loggingDB.escapeString(referrer);
+    // TODO: Refactor to corresponding webext logic or discard
+    // let referrer = "";
+    // if (details.referrer) {
+    //   referrer = details.referrer.spec;
+    // }
+    // update.referrer = escapeString(referrer);
 
-  const current_time = new Date();
-  update.time_stamp = current_time.toISOString();
+    const current_time = new Date(details.timeStamp);
+    update.time_stamp = current_time.toISOString();
 
-  let encodingType = "";
-  const headers = [];
-  let isOcsp = false;
-  httpChannel.visitRequestHeaders({
-    visitHeader(name, value) {
+    let encodingType = "";
+    const headers = [];
+    let isOcsp = false;
+    details.requestHeaders.map(requestHeader => {
+      const { name, value } = requestHeader;
       const header_pair = [];
-      header_pair.push(loggingDB.escapeString(name));
-      header_pair.push(loggingDB.escapeString(value));
+      header_pair.push(escapeString(name));
+      header_pair.push(escapeString(value));
       headers.push(header_pair);
       if (name === "Content-Type") {
         encodingType = value;
         if (encodingType.indexOf("application/ocsp-request") !== -1) {
           isOcsp = true;
         }
       }
-    },
-  });
-
-  if (requestMethod === "POST" && !isOcsp) {
-    // don't process OCSP requests
-    reqEvent.subject.QueryInterface(components.interfaces.nsIUploadChannel);
-    if (reqEvent.subject.uploadStream) {
-      reqEvent.subject.uploadStream.QueryInterface(
-        components.interfaces.nsISeekableStream,
-      );
-      const postParser = new httpPostParser.HttpPostParser(
-        reqEvent.subject.uploadStream,
-        loggingDB,
-      );
-      const postObj = postParser.parsePostRequest(encodingType);
-
-      // Add (POST) request headers from upload stream
```

Comment:
@englehardt: I am not familiar with upload streams with headers, but they may potentially be unsupported: https://bugzilla.mozilla.org/show_bug.cgi?id=1416486 - I'll leave this code as is for now.